File: sdk/csharp/openai-chat/ChatService.cs
using OpenAI;
using OpenAI.Chat;

namespace GonkaGate.Examples;

internal static class ChatService
{
    private const string SystemPrompt = "You are a concise assistant.";

    public static string Complete(AppConfig config, string prompt)
    {
        var options = new OpenAIClientOptions
        {
            Endpoint = new Uri(config.BaseUrl)
        };

        var client = new ChatClient(config.Model, config.ApiKey, options);

        var completion = client.CompleteChat(
            [
                new SystemChatMessage(SystemPrompt),
                new UserChatMessage(prompt)
            ],
            new ChatCompletionOptions
            {
                Temperature = 0.2f
            }
        );

        var text = string.Concat(
            completion.Content
                .Where(contentPart => !string.IsNullOrWhiteSpace(contentPart.Text))
                .Select(contentPart => contentPart.Text!.Trim())
        );

        if (string.IsNullOrWhiteSpace(text))
        {
            throw new InvalidOperationException("Received a response but no message content.");
        }

        return text;
    }
}

File: sdk/java/openai-chat/src/main/java/com/gonkagate/examples/ChatService.java
package com.gonkagate.examples;

import com.openai.client.OpenAIClient;
import com.openai.client.okhttp.OpenAIOkHttpClient;
import com.openai.models.chat.completions.ChatCompletion;
import com.openai.models.chat.completions.ChatCompletionCreateParams;

final class ChatService {
    private ChatService() {
    }

    static String complete(AppConfig config, String systemPrompt, String userPrompt) {
        OpenAIClient client = OpenAIOkHttpClient.builder()
            .apiKey(config.apiKey())
            .baseUrl(config.baseUrl())
            .build();

        ChatCompletionCreateParams request = ChatCompletionCreateParams.builder()
            .model(config.model())
            .addSystemMessage(systemPrompt)
            .addUserMessage(userPrompt)
            .build();

        ChatCompletion response = client.chat().completions().create(request);
        String text = extractText(response);

        if (Strings.isBlank(text)) {
            throw new IllegalStateException("Received a response but no message content in choices[0].");
        }

        return text;
    }

    private static String extractText(ChatCompletion response) {
        return response.choices().stream()
            .findFirst()
            .flatMap(choice -> choice.message().content())
            .map(String::trim)
            .orElse("");
    }
}
